pyramid_zipkin/tween.py

```diff
@@ -73,8 +75,14 @@ def _get_settings_from_request(request):
 
     if 'zipkin.transport_handler' in settings:
         transport_handler = settings['zipkin.transport_handler']
-        stream_name = settings.get('zipkin.stream_name', 'zipkin')
-        transport_handler = functools.partial(transport_handler, stream_name)
+        if not isinstance(transport_handler, BaseTransportHandler):
+            warnings.warn(
+                'Using a function as transport_handler is deprecated. '
```

Comment:
You don't know for sure that it's a function. Check `transport_handler`'s`type` first? And error out if it's not a function.

Reply:
This is mostly to keep backwards compatibility. Our idea was to deprecate using a function for now and then after a couple of months remove support for it entirely.  Would be also nice to release a 1.0 version at some point...

Reply:
If you need to break compatibility eventually, then it makes sense to do a 1.0 release now. No real advantage to doing a deprecation period IMO.

But a question worth asking is: do you actually need to break compatibility or is it possible to keep supporting both function and class based transport handlers? If it's possible I'd favor not breaking compatibility.

Reply:
I think breaking backwards compatibility is the right thing to do here. Using a transport handler type is a cleaner approach in the long run IMO. We could immediately break backwards compatibility, but i'd prefer to give a little warning (see my comment above)